numpy/ma/core.py

```diff
@@ -3493,6 +3493,10 @@ def _get_flat(self):
     def _set_flat(self, value):
         "Set a flattened version of self to value."
         y = self.ravel()
+        # TODO: This is messed up, since there is no guarantee for
+        # ravel to return views into either array or mask.
+        # If self._sharedmask, we still do want self._mask to change here:
+        y._sharedmask = False
```

Comment:
I don't know if I should leave this, the result changing based on sharedmask seems nonsense, but it does change behaviour once in a while. On the other hand we would have to futurewarn this on its own.